### PR TITLE
Feature/add api tests

### PR DESCRIPTION
**Pull request by kemal-rov**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,19 +16,30 @@
...
     "@types/lodash": "^4.14.202",
     "@types/mongoose": "^5.11.97",
+    "axios-cookiejar-support": "^5.0.0",
     "body-parser": "^1.20.2",
     "compression": "^1.7.4",
     "cookie-parser": "^1.4.6",
     "cors": "^2.8.5",
     "dotenv": "^16.4.2",
     "express": "^4.18.2",
+    "jest": "^29.7.0",
```
Comment (sepisoltani) [inline package.json]: The jest package should generally be listed as a development dependency in your project.

File: package.json [hunk 2]
```diff
@@ -16,19 +16,30 @@
...
     "lodash": "^4.17.21",
     "mongodb": "^6.3.0",
     "mongoose": "^8.1.1",
     "nodemon": "^3.0.3",
+    "tough-cookie": "^4.1.3",
+    "ts-jest": "^29.1.2",
```
Comment (sepisoltani) [inline package.json]: the ts-jest package should generally be listed as a development dependency in your project.

File: src/__tests__/authentication.test.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        throw error; // If it's not an AxiosError
+      }
+    }
+  });
+
+  it('should reject login with incorrect password', async () => {
+    try {
+      await axiosInstance.post(`${url}/auth/login`, {
+        email: newEmail,
+        password: 'wrongpassword', // Incorrect password
```
Comment (sepisoltani) [inline src/__tests__/authentication.test.ts]: You have an extra comment here. The `wrongpassword` is explaining itself

File: src/__tests__/authentication.test.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    expect(loginResponse.status).toBe(200);
+    expect(loginResponse.data).toHaveProperty('email', newEmail);
+  });
+
+  it('should logout the user', async () => {
+    // Ensure there's a session to logout from by logging in first
+    await axiosInstance.post(`${url}/auth/login`, {
+      email: newEmail,
+      password: password,
+    });
```
Comment (sepisoltani) [inline src/__tests__/authentication.test.ts]: You can simplify this code:

```javascript

await axiosInstance.post(`${url}/auth/login`, { email: newEmail, password });


```

File: package.json (hunk 1 above)
Comment (kemal-rov) [reply package.json]: Great observation, I'll reinstall the packages properly

File: package.json (hunk 2 above)
Comment (kemal-rov) [reply package.json]: This as well

File: src/__tests__/authentication.test.ts (hunk 4 above)
Comment (kemal-rov) [reply src/__tests__/authentication.test.ts]: Very ES6 friendly, I like it! On it

**Pull request merged**